src/nodes.coffee

```diff
@@ -1362,6 +1380,10 @@ exports.Class = class Class extends Base
         @ctor = method
       else if method.isStatic and method.bound
         method.context = @name
+      else if method.bound
+        @boundMethods.push method.name
+        @setParentRef(o)
```

Comment:
if we have any bound methods, see if the parent class expression should be cached to a variable ref and save the (possibly updated) parent on `method` so that it can be used in the runtime check